crates/nixci/src/command/run.rs

```diff
@@ -76,6 +92,70 @@ impl RunCommand {
         Ok(())
     }
 
+    /// Run the ci run steps on remote
+    async fn run_remote(&self, nixcmd: &NixCmd, cfg: Config, host: &str) -> anyhow::Result<()> {
+        let metadata = FlakeMetadata::from_nix(nixcmd, &cfg.ref_.flake_url).await?;
+
+        let omnix_source = PathBuf::from(OMNIX_SOURCE);
+
+        nix_rs::copy::nix_copy(nixcmd, host, &[omnix_source.clone(), metadata.path.clone()])
+            .await?;
+
+        let ci_run_args = self.get_ci_run_args_for_remote(metadata.path, cfg.ref_)?;
+
+        let nix_run_args: Vec<String> = vec![
+            "nix run".to_string(),
+            format!("{}#default", OMNIX_SOURCE),
+            "--".to_string(),
+        ]
+        .into_iter()
+        .chain(ci_run_args.into_iter())
+        .collect();
+
+        // call ci run on remote machine through ssh
+        ssh::on_ssh(host, &nix_run_args).await?;
+
+        Ok(())
+    }
+
+    // Return ci run args along with build_step_args
+    fn get_ci_run_args_for_remote(
```

Comment:
I am still doubting that this would be the correct way to go about it. Little advice would be appreciated.